### DISCUSSION TITLE
How to execute a sql script without opening it ? Is this even possible?

### DISCUSSION THREAD

**Discussion by emanueol** [dbeaver/dbeaver#35922] [Q&A] [answered]
I tried using @script filename but doesnt work:
![image](https://github.com/user-attachments/assets/bda9e563-009e-4dad-bbc8-35c6a5699d66)

Error is:
```
SQL Error [1003] [42000]: SQL compilation error:
syntax error line 1 at position 0 unexpected '@tablepermanent.sql'.
```

FYI: Im connected to a Snowflake database.

Probably I need to use snow or snowsql cli for this, but anyway, just trying to know if dbeaver itself can execute an external file without using cliky-cliky-mouse-on-gui

Thanks ~

**Comment by TB-HunBoka** [comment]
Yes you can, but I realized that it is bugged for me.
However, it is the `@include` command you are looking for.
Example: 
@include runsqltest.sql
@include C:\...fullpath...\Scripts\runsqltest.sql

For me it runs the query, I can see the results for a second, then the tab is closed and overriden by the statistics tab.

Can you check it too? If you experience the same, then open a bug ticket for it.

**Reply by TB-HunBoka** [reply]
Okay, I changed something and now it opens a new console for each `@include `commands.

**Reply by emanueol** [reply]
Doesnt work for me.. see screenshot, i got 2 sql scripts in 1 folder and just want open script 1 > select and execute line that executes a 2nd script:
![image](https://github.com/user-attachments/assets/bc210c08-0a4b-4209-921a-5c3011726607)

FYI:
1. Im not using DBEaver workspaces, I always save and edit in DBEaver scripts directly from folders.
2. Im aware that we can link folders inside DBeaver:
![image](https://github.com/user-attachments/assets/f2e5a31f-07a4-48ba-a782-dd565f63a8fa)
3. Unsure if that's needed to avoid the error that it cant find 2nd script **DESPITE ITS ON SAME FOLDER** of 1st script containing the @include

Anyway, this is not critical because for me testing how to do this is over after I see dbeaver unable to find 2nd script despite both scripts in same folder (maybe someone can open an improvement).

My goal was simple to avoid the need to execute outside dbeaver my 1st script, so I can just edit all SQL scripts inside dbeaver and execute them, but.. no problem, I will shell execute snowsql it:
![image](https://github.com/user-attachments/assets/b32b6896-8d9a-4e4c-ba99-2acff515c3c1)

**Comment by emanueol** [comment]
Thanks, will try later and feedback.

Extra thingie, snowsql (snowflake cli) uses exclamation to edecute
secondary scripts, i wonder if theres a way to configure kind of alias in
dbeaver do dbeaver interprets:
!myscript.sql

as
@include myscript.sql

Best Regards,
Emanuel Oliveira

**Reply by TB-HunBoka** [reply]
Never used this option, I know that !source will execute the sql query in snowflake, but I am not sure what do you mean "configure kind of alias".

The `@include myscript.sql` command in DBeaver will read the sql file, and execute it in a sql console as much as I know.

I might not understand what your goal is with running a sql file from editor, but are you sure this is the best way to get what you would like to achieve?
Are these repetitive things? -> do a database task
You don't want to type in the same query all the time? -> create sql template for it

**Comment by TB-HunBoka** [comment, marked as answer]
Oh, I see what went wrong. The DBeaver command doesn't requires the semicolon after the file name. The DBeaver could not find a filename with that character included.
You can run it with this code:
`@include tablepermanent.sql`

It will find the file in the same folder as the primary sql, no need to be saved in the workspace folder.

**Reply by emanueol** [reply]
Thanks finally we got it :)
To execute external scripts we use:

```sql
@include tablepermanent.sql
```

Good that's possible, but seems weak and very rudimentary support:
1. Dbeaver opens the script mentioned on the @include > executes it > but.. theres no output of script execution for example.. very bad.
2. ![dbeaver--execute external script via @include](https://github.com/user-attachments/assets/4ad6ca43-32dc-4a40-ba64-34e841d8b854)
3. and after some editor contents replacement happens every time we now press CTRL+S dbeaver 
   starts asking if we want to replace editor contents. Need to exit DBeaver and relaunch for it
   to stop doing that. I suspect this is from another [issue with Google Drive](https://github.com/dbeaver/dbeaver/issues/16952) I opened in the past.
![dbeaver--execute external script via @include CTRL+S popup](https://github.com/user-attachments/assets/73227cb9-8d40-4102-a2ae-adaab9249de8)

So now, I wonder if possible to retain execution output of external script ?

**Reply by ask9** [reply]
>     1. Dbeaver opens the script mentioned on the @include > executes it > but.. theres no output of script execution for example.. very bad.

I recommend to file a bug report: https://github.com/dbeaver/dbeaver/issues

**Reply by emanueol** [reply]
> > ```
> > 1. Dbeaver opens the script mentioned on the @include > executes it > but.. theres no output of script execution for example.. very bad.
> > ```
> 
> I recommend to file a bug report: https://github.com/dbeaver/dbeaver/issues

Nah, will not pursue this one anymore, thanks for your help.. will mark it closed with the solution you provided:
```sql
@include external.sql
```

Thanks ~

**Reply by TB-HunBoka** [reply]
> > > ```
> > > 1. Dbeaver opens the script mentioned on the @include > executes it > but.. theres no output of script execution for example.. very bad.
> > > ```

This is the same thing I mentioned earlier. You can change this behavior to keep the executed script window opened in a new tab by disabling the "Close included script tab after execution":
![kép](https://github.com/user-attachments/assets/d1ef3097-075d-40f1-b478-974f59301e83)

**Reply by ask9** [reply]
Thanks for the option.
Unfortunately, it opens a new tab with the script file itself (not result tab in the same sql editor tab), what makes it difficult to use for me (but maybe it is suitable for @emanueol).
I would prefer to get results tab from the script in the same tab where `@include` was called from.
